src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -710,7 +710,7 @@ private async Task<ConnectBlocksResult> ConnectChainAsync(ChainedHeader newTip,
                         {
                             uint256 newFinalizedHash = blockToConnect.ChainedHeader.GetAncestor(newFinalizedHeight).HashBlock;
 
-                            await this.finalizedBlockInfo.SaveFinalizedBlockHashAndHeightAsync(newFinalizedHash, newFinalizedHeight).ConfigureAwait(false);
+                            this.finalizedBlockInfo.SaveFinalizedBlockHashAndHeight(newFinalizedHash, newFinalizedHeight);
```

Comment:
Could this piece of code go in to a rule?

Reply:
Yeah, sounds like a good idea. Will add to my todo list

Reply:
In fact this should be part of the CoinView right now, becuse of flushing.
There is a race condition if node will crush and coin view did not persist.

Reply:
Good point